src/driver/mpas_subdriver.F

```diff
@@ -109,38 +111,68 @@ end subroutine xml_stream_get_attributes
       readNamelistArg = .false.
       readStreamsArg = .false.
 
-      nArgs = command_argument_count()
-      iArg = 1
-      do while (iArg < nArgs)
-         call get_command_argument(iArg, argument)
-         if (len_trim(argument) == 0) exit
-
-         if ( trim(argument) == '-n' ) then
-            iArg = iArg + 1
+      ! If provided, error check the namelistFileParam and copy it to namelistFile to override default
+      if (present(namelistFileParam)) then
+         if (len_trim(namelistFileParam) <= 0) then
+            write(0,'(A)') 'WARNING: mpas_init argument ''namelistFileParam'' has 0 or negative length and will be ignored'
+         else if (len_trim(namelistFileParam) > len(namelistFile)) then
+            write(0,'(A,I5,A,I5,A)') 'CRITICAL ERROR: mpas_init argument ''namelistFileParam'' has length ', len_trim(namelistFileParam),&
+                                     ', but the maximum allowed is ', len(namelistFile), ' characters'
+            stop
+         else
             readNamelistArg = .true.
-            call get_command_argument(iArg, namelistFile)
-            if ( len_trim(namelistFile) == 0 ) then
-                write(0,*) 'ERROR: The -n argument requires a namelist file argument.'
-                stop
-            else if ( trim(namelistFile) == '-s' ) then
-                write(0,*) 'ERROR: The -n argument requires a namelist file argument.'
-                stop
-            end if
-         else if ( trim(argument) == '-s' ) then
-            iArg = iArg + 1
+            namelistFile = trim(namelistFileParam)
+         endif
+      endif
+      ! If provided, error check the streamsFileParam and copy it to streamsFile to override default
+      if (present(streamsFileParam)) then
+         if (len_trim(streamsFileParam) <= 0) then
+            write(0,'(A)') 'WARNING: mpas_init argument ''streamsFileParam'' has 0 or negative length and will be ignored'
```

Comment:
I forgot to add this to my previous review, but I don't think `len_trim` will ever return a negative number, so perhaps mentioning (and testing for) a negative length here may be unnecessarily confusing?
Here's what the Fortran 2003 standard has to say about the return value of `len_trim`:
```
The result has a value equal to the number of characters remaining after any
trailing blanks in STRING are removed. If the argument contains no nonblank characters, the
result is zero.
```

Reply:
This just came from me writing my if tests to be really exact. I agree it would be better without the negative length check/print confusing the issue.

Reply:
In general I agree that writing conditional statements so that they completely cover all possibilities is the correct thing to do. I suppose (not that we need to make yet another amendment for this PR) we could have tested whether `len_trim(streamsFileParam) <= 0`, but have the write statement say something less algebraic, e.g., "streamsFileParam is empty and will be ignored.".